kvs/src/kvs/server.cpp

```diff
@@ -587,6 +585,37 @@ void run(unsigned thread_id, Address public_ip, Address private_ip,
 
       report_start = std::chrono::system_clock::now();
 
+      // Get the most recent list of cache IPs.
+      // (Actually gets the list of all current functional nodes.)
+
+      zmq::socket_t func_nodes_requester(context, ZMQ_REQ);
```

Comment:
Nit: We probably don't need to recreate this every time, since the IP/port combination doesn't change. Should be able to create it once.

Reply:
Got it; socket connection happens at beginning of run() now.